Fix/code validation min length

AdrianCoello opened the pull request:
This pull request standardizes and strengthens the validation logic for the `code` field across student and professor forms. The `code` field is now consistently treated as a string, with centralized validation constants and improved user feedback for invalid inputs. Additionally, the UI is updated to enforce minimum and maximum digit constraints, and extra debugging/logging is added for development purposes.

**Validation Logic Improvements:**

* The `code` field is now validated as a string using a shared `CODE_REGEX` that enforces a minimum of 5 and a maximum of 8 digits, with error messages centralized in `CODE_ERROR_MESSAGE`. This applies to all relevant forms for students and professors. [[1]](diffhunk://#diff-fc185cfa54760e094cc7dc6edc934b0b9c760a6eb6d71ad73a2932e27712f788R2-R12) [[2]](diffhunk://#diff-a47b4c6510fa6331707d394fc82dbf4da409a9ed94641d3cc255caf25c8218f1L29-R64) [[3]](diffhunk://#diff-93f1a1e64ed245b894267d0b3f13d7060c87c3198f0184dc90eef439d05e804fL18-R31) [[4]](diffhunk://#diff-6bcc4fcf20fa18a95d79c2b9342919aea83df1c4b18738c000dafb62101b1d97L34-R39) [[5]](diffhunk://#diff-4271403b7691f028e9193195c0ba9c11aeb0b37a6aa46479703faa35a5a4e062L30-R43) [[6]](diffhunk://#diff-a840b738a1470129b4b671433238ed66f58d79b334602aef685aacc393347d8eR58-R63) [[7]](diffhunk://#diff-5e313aa10307536353547fbc27270d34139e52bc7dbfca48a2685cabaf0f7f30L55-R57)
* Validation constants (`CODE_ERROR_MESSAGE`, `CODE_DIGITS`, `CODE_MIN_DIGITS`, `CODE_REGEX`) are imported and used throughout the codebase for consistency. [[1]](diffhunk://#diff-a47b4c6510fa6331707d394fc82dbf4da409a9ed94641d3cc255caf25c8218f1R18-R23) [[2]](diffhunk://#diff-93f1a1e64ed245b894267d0b3f13d7060c87c3198f0184dc90eef439d05e804fR7-R12) [[3]](diffhunk://#diff-6bcc4fcf20fa18a95d79c2b9342919aea83df1c4b18738c000dafb62101b1d97R11-R17) [[4]](diffhunk://#diff-4271403b7691f028e9193195c0ba9c11aeb0b37a6aa46479703faa35a5a4e062R12-R24) [[5]](diffhunk://#diff-5e313aa10307536353547fbc27270d34139e52bc7dbfca48a2685cabaf0f7f30R21-R23) [[6]](diffhunk://#diff-a840b738a1470129b4b671433238ed66f58d79b334602aef685aacc393347d8eR17-R22)

**UI and Form Handling Updates:**

* The `inputProps` for the `code` field are updated to use `maxLength` and `minLength` based on the new validation constants, ensuring that users cannot enter codes outside the allowed range. [[1]](diffhunk://#diff-a47b4c6510fa6331707d394fc82dbf4da409a9ed94641d3cc255caf25c8218f1L188-R285) [[2]](diffhunk://#diff-93f1a1e64ed245b894267d0b3f13d7060c87c3198f0184dc90eef439d05e804fL162-R179) [[3]](diffhunk://#diff-6bcc4fcf20fa18a95d79c2b9342919aea83df1c4b18738c000dafb62101b1d97L171-R183) [[4]](diffhunk://#diff-4271403b7691f028e9193195c0ba9c11aeb0b37a6aa46479703faa35a5a4e062L199-R226) [[5]](diffhunk://#diff-5e313aa10307536353547fbc27270d34139e52bc7dbfca48a2685cabaf0f7f30L213-R222)
* The submit button is now disabled if the form is invalid or contains errors, preventing accidental submission of invalid data.

**Code and Data Handling Adjustments:**

* All form submissions and update operations now treat the `code` as a string, aligning with the new validation logic and ensuring consistent data handling. [[1]](diffhunk://#diff-6bcc4fcf20fa18a95d79c2b9342919aea83df1c4b18738c000dafb62101b1d97L68-R77) [[2]](diffhunk://#diff-4271403b7691f028e9193195c0ba9c11aeb0b37a6aa46479703faa35a5a4e062L85-R116) [[3]](diffhunk://#diff-5e313aa10307536353547fbc27270d34139e52bc7dbfca48a2685cabaf0f7f30L108-R116)
* The `handleCodeChange` logic in forms is updated to immediately validate and set errors for the `code` field as the user types, and to mark the field as touched for better feedback. [[1]](diffhunk://#diff-a47b4c6510fa6331707d394fc82dbf4da409a9ed94641d3cc255caf25c8218f1R173-R203) [[2]](diffhunk://#diff-93f1a1e64ed245b894267d0b3f13d7060c87c3198f0184dc90eef439d05e804fR97) [[3]](diffhunk://#diff-4271403b7691f028e9193195c0ba9c11aeb0b37a6aa46479703faa35a5a4e062R145)

AdrianCoello committed:
feat: add minimum length validation for student and professor codes

AdrianCoello committed:
feat: implement minimum 5-character validation for student and professor codes

- Add CODE_MIN_DIGITS=5 and CODE_ERROR_MESSAGE constants
- Update CODE_REGEX to enforce 5-8 digit requirement
- Implement validation in all student and professor forms (create/edit)
- Fix backend communication by filtering required/unwanted fields

IvanTicona commented on src/components/common/EditProfessorComponent.tsx:
```diff
@@ -26,10 +32,36 @@ const validationSchema = Yup.object({
-  code: Yup.number().optional(),
+  code: Yup.string()
+    .test('code-validation', CODE_ERROR_MESSAGE, function(value) {
+      console.log('=== VALIDACIÓN SCHEMA YUP ===');
```
Todos estos logs no deberian llegar a tu PR

IvanTicona commented on src/components/common/EditProfessorComponent.tsx:
```diff
@@ -104,9 +170,37 @@ const EditProfessorComponent: React.FC<{ id: number; onClose: () => void }> = ({
       formik.setFieldValue("code", value);
+      formik.setFieldTouched("code", true); 
+      
+      console.log('Validando inmediatamente...');
```
Mas logs que no deberian llegar a tu PR

AdrianCoello committed:
refactor: remove console.log statements from EditProfessorComponent

- Remove debug console.log statements for production readiness
- Clean up validation logging and form change tracking logs
- Maintain functionality while removing debugging artifacts

AdrianCoello committed:
fix: address Copilot feedback on code validation PR

- Remove debugging console.log statements from EditProfessorPage
- Improve error handling in EditProfessorComponent with proper user feedback
- Simplify button disabled condition removing redundant validation check
- Remove non-existent username and password properties from form submissions
- Fix updateProfessor call by adding missing id parameter
